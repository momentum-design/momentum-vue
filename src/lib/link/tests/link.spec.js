import { shallowMount } from '@vue/test-utils';
import Link from '../index.vue';

describe('Link', () => {
  it('should match snapshot', () => {
    const container = shallowMount(Link, {
      slots: {
        default: `<div class="child" />`
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render class if prop is passed', () => {
    const container = shallowMount(Link, {
      attrs: {
        class: 'dark'
      },
      slots: {
        default: `<div class="child" />`
      }
    });

    expect(container.find('.dark').exists()).toBeTruthy();
  });


  it('should render color if prop is passed', () => {
    const container = shallowMount(Link, {
      propsData: {
        color: 'green'
      },
      slots: {
        default: `<div class="child" />`
      }
    });

    expect(container.find('.md-link--green').exists()).toBeTruthy();
  });

  it('should render theme if prop is passed', () => {
    const container = shallowMount(Link, {
      propsData: {
        theme: 'dark'
      },
      slots: {
        default: `<div class="child" />`
      }
    });

    expect(container.find('.md-link--dark').exists()).toBeTruthy();
  });

  it('should render disabled if prop is passed', () => {
    const container = shallowMount(Link, {
      propsData: {
        disabled: true
      },
      slots: {
        default: `<div class="child" />`
      }
    });

    expect(container.attributes().disabled).toEqual('disabled');
  });


  it('should render children', () => {
    const container = shallowMount(Link, {
      slots: {
        default: `<div class="child" />`
      }
    });

    expect(container.find('.child').exists()).toBeTruthy();
  });
});

