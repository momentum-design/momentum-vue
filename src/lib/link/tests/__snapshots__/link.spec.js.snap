// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<a tabindex="0" class="md-link md-link--blue">
  <div class="child"></div>
</a>
`;
